coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -3357,6 +3384,13 @@ def to(context, node):
             "Received invalid arguments for PyTorch conversion of op {}".format(node)
         )
 
+    # We have to handle the case where the dtype is not set, this should be inferred from the Tensor dtype
+    # see, https://pytorch.org/docs/stable/generated/torch.Tensor.to.html?highlight=#torch.Tensor.to
+    if dtype is None:
+        out = mb.identity(x=_input, name=node.name)
+        context.add(out, node.name)
+        return = 0 # TODO: infer from Tensor (spoiler in this case we care about its f32 => 6)
```

Comment:
This might make sense for your model but I don't think this makes sense in general. Probably want to do the TODO or make the body of this if-statement raise an exception.

Reply:
My bad, looks like a bad typo there and I left in a TODO 😵‍💫. 

This should be 
```
if dtype is None:
        out = mb.identity(x=_input, name=node.name)
        context.add(out, node.name)
        return
```
that is if the dtype to convert the tensor to is None, then  the inferred dtype is that of the input tensor. Here I then just make a copy of the input tensor and return early. I think this is aligned with the torch docs linked (https://pytorch.org/docs/stable/generated/torch.Tensor.to.html?highlight=#torch.Tensor.to)

Does that make sense?

Reply:
That makes sense.